apps/linode-marketplace-wazuh/roles/wazuh/tasks/main.yml

```diff
@@ -0,0 +1,81 @@
+---
+# nginx
+- name: install nginx
+  apt:
+    pkg: nginx
+    state: present
+
+- name: config file setup
+  template:
+      src: templates/nginx.conf.j2
+      dest: "/etc/nginx/sites-available/{{ _domain }}"  
+
+- name: create a symbolic link for the nginx configuration
+  file:
+    src: "/etc/nginx/sites-available/{{ _domain }}"
+    dest: "/etc/nginx/sites-enabled/{{ _domain }}"
+    state: link
+
+- name: unlink the default nginx configuration
+  file:
+    path: "/etc/nginx/sites-enabled/default"
+    state: absent
+  notify: reload nginx service
+
+# ssl_certbot
+- name: setting up ssl certificate
+  import_role:
+    name: certbot_ssl
+
+# wazuh
+- name: Download Wazuh install script
+  ansible.builtin.get_url:
+    url: https://packages.wazuh.com/4.7/wazuh-install.sh
+    dest: /tmp/wazuh-install.sh
+    mode: '0755'
+
+- name: Stop and disable nginx service before Wazuh installation
+  systemd:
+    name: nginx
+    state: stopped
+    enabled: no
+
+- name: Install Wazuh
+  ansible.builtin.shell:
+    cmd: ./wazuh-install.sh -a
+    chdir: /tmp
+
+- name: Extract the entire Wazuh installation files tar
+  ansible.builtin.unarchive:
+    src: /tmp/wazuh-install-files.tar
+    dest: /tmp/
+    remote_src: yes
+
+- name: Read content from /tmp/wazuh-install-files/wazuh-passwords.txt
+  ansible.builtin.slurp:
+    src: /tmp/wazuh-install-files/wazuh-passwords.txt
+  register: wazuh_passwords_content
+
+- name: decode and append wazuh_passwords_content to creds file
+  ansible.builtin.shell:
+    cmd: |
+      echo "" >> /home/{{ username }}/.credentials
+      echo "{{ wazuh_passwords_content['content'] | b64decode }}" >> /home/{{ username }}/.credentials
```

Comment:
This needs to be reworked a bit to avoid using `shell`. We can use `lineinfile` to insert before EOF

Reply:
`blockinfile` has been used, since the wazuh-passwords.txt file is a whole mess of UNs and PWs.
```
- name: append decoded wazuh_passwords_content to credentials file
  ansible.builtin.blockinfile:
    path: /home/{{ username }}/.credentials
    block: "{{ wazuh_passwords_content['content'] | b64decode }}"
    marker: "###"
    create: yes
    owner: "{{ username }}"
    group: "{{ username }}"
    mode: '0600'
```